source/common/common/macros.h

```diff
@@ -16,4 +16,17 @@ namespace Envoy {
  * Stop the compiler from complaining about an unreferenced parameter.
  */
 #define UNREFERENCED_PARAMETER(X) ((void)(X))
+
+/**
+ * Have a generic fall-through for different versions of C++
+ */
+#if __cplusplus >= 201402L // C++14, C++17 and above
+#define FALLTHRU [[fallthrough]]
+#elif __cplusplus >= 201103L && __GNUC__ == 7 // C++11 gcc 7
+#define FALLTHRU [[gnu::fallthrough]]
+#elif __cplusplus >= 201103L // C++11 gcc 6 and older
+#define FALLTHRU
+#else // C and C++03
+#define FALLTHRU __attribute__ ((fallthrough))
```

Comment:
On second thought, should we just change this to an #error? I don't think this should ever happen.